src/ILVerification/src/TypeVerifier.cs

```diff
@@ -74,31 +76,38 @@ public void VerifyInterfaces()
                     InterfaceImplementationHandle = interfaceHandle
                 };
 
-                // Look for duplicates.
                 if (!implementedInterfaces.Contains(imo))
                 {
                     implementedInterfaces.Add(imo);
                 }
                 else
                 {
-                    VerificationError(VerifierError.InterfaceImplHasDuplicate, type, interfaceType, _module.MetadataReader.GetToken(interfaceHandle));
+                    VerificationError(VerifierError.InterfaceImplHasDuplicate, Format(type, interfaceType));
                 }
+            }
 
+            foreach (InterfaceMetadataObjects implementedInterface in implementedInterfaces)
+            {
                 if (!type.IsAbstract)
-                { 
+                {
                     // Look for missing method implementation
-                    foreach (MethodDesc method in imo.DefType.GetAllMethods())
+                    foreach (MethodDesc method in implementedInterface.DefType.GetAllMethods())
                     {
                         MethodDesc resolvedMethod = virtualMethodAlg.ResolveInterfaceMethodToVirtualMethodOnType(method, type);
                         if (resolvedMethod is null)
                         {
-                            VerificationError(VerifierError.InterfaceMethodNotImplemented, type, imo.DefType, method, _module.MetadataReader.GetToken(_typeDefinitionHandle), _module.MetadataReader.GetToken(imo.InterfaceImplementationHandle), _module.MetadataReader.GetToken(((EcmaMethod)method).Handle));
+                            VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type, implementedInterface.DefType, method));
```

Comment:
I think it would be better to call format on each of the elements. Something like:

`VerificationError(VerifierError.InterfaceMethodNotImplemented, Format(type), Format(implementedInterface.DefType), Format(method))`

Otherwise, we would need many different `Format` signatures for different combinations of types as the variety of error messages grows.

Reply:
Funny my first idea was like that, but code looked strange :smile:
I'll fix It ASAP thank's!

Reply:
@jkotas added helper and some comments!